lib/cell.rb

```diff
@@ -17,4 +18,12 @@ def place_ship(ship_object_parameter)
   def fired_upon?
     false
   end
+
+  def fire_upon
+    if @empty == true
```

Comment:
If we change .empty? as suggested above, we can change this line to use it:
`if empty?`
(Also, if we _didn't_ change it back, you could simplify this to:
`if @empty`
...because any expression or variable that you know holds either `true` or `false` doesn't need to be compared (with ==) to `true` or `false`, because by itself its return value will either be true or false.)